src/frontier_util.c

```diff
@@ -83,16 +94,74 @@ static void ShowPyramidResultsWindow(void);
 static void ShowLinkContestResultsWindow(void);
 static void CopyFrontierBrainText(bool8 playerWonText);
 
-// const rom data
-static const u8 sFrontierBrainStreakAppearances[NUM_FRONTIER_FACILITIES][4] =
+#define FRONTIER_BRAIN_SPRITES(Brain) \
+    .trainerId = TRAINER_##Brain,     \
+    .objEventGfx = OBJ_EVENT_GFX_##Brain
+
+#define FRONTIER_BRAIN_TEXTS(Brain)                                        \
+    .lostTexts = {gText_##Brain##DefeatSilver, gText_##Brain##DefeatGold}, \
+    .wonTexts = {gText_##Brain##WonSilver, gText_##Brain##WonGold}
+
+// battledBit: Flags to change the conversation when the Frontier Brain is encountered for a battle
+// First bit is has battled them before and not won yet, second bit is has battled them and won (obtained a Symbol)
+const struct FrontierBrain gFrontierBrainInfo[NUM_FRONTIER_FACILITIES] =
 {
-    [FRONTIER_FACILITY_TOWER]   = {35,  70, 35, 1},
-    [FRONTIER_FACILITY_DOME]    = { 4,   9,  5, 0},
-    [FRONTIER_FACILITY_PALACE]  = {21,  42, 21, 1},
-    [FRONTIER_FACILITY_ARENA]   = {28,  56, 28, 1},
-    [FRONTIER_FACILITY_FACTORY] = {21,  42, 21, 1},
-    [FRONTIER_FACILITY_PIKE]    = {28, 140, 56, 1},
-    [FRONTIER_FACILITY_PYRAMID] = {21,  70, 35, 0},
+    [FRONTIER_FACILITY_TOWER] =
+    {
+        FRONTIER_BRAIN_SPRITES(ANABEL),
+        .isFemale = TRUE,
+        FRONTIER_BRAIN_TEXTS(Anabel),
+        .battledBit = {1 << 0, 1 << 1},
```

Comment:
I noticed that the battledBit values end being `1 << ((2 * facility) + hasSymbol)`

Unsure if that's useful in any way